File: expensetracker-frontend/src/api/AuthService.js
import axios from "axios";
import { API_URL } from "./constants";
export default class AuthService{

    signin(body){
        return axios.post(API_URL+"/Account/SignIn", body);
    }

    confirm(body){
        return axios.post(API_URL+"/Account/Confirm","token="+body);
    }

    register(body){
        return axios.post(API_URL+"/Account/Register", body);
    }

    forgotPassword(emailAddress){
        return axios.post(API_URL+"/Account/ForgotPassword", "email="+emailAddress);
    }
    resetPassword(Token,Password){
        return axios.post(API_URL+"/Account/ResetPassword", {Token,Password});
    }
}

File: mobile/fisk/lib/data/services/auth_service.dart
import 'dart:convert';

import 'package:dio/dio.dart';
import 'package:fisk/data/models/login/loginResponse/login_response.dart';
import 'package:fisk/data/services/BaseApi.dart';

class AuthService{
    Future<LoginResponse> signin({required String email, required String password}) async {
        var params =  {
            "emailAddress": email,
            "Password": password,
        };
        var response = await Dio().post("${BaseApi.productionURL}/Account/SignIn",data:jsonEncode(params));
        //print(response);
        final responseJson = jsonDecode(response.toString());
        //print(LoginResponse.fromJson(responseJson).token);
        return LoginResponse.fromJson(responseJson);

    }
}

